enrichment-web/src/main/java/eu/europeana/enrichment/web/service/impl/EnrichmentTopicServiceImpl.java

```diff
@@ -103,13 +121,108 @@ public Topic deleteTopic(String topicIdentifier) {
 		return dbtopicEntity;
 	}
 
-	@Override
-	public String searchTopics(String query, String fq, String fl, String facets, String sort, int page, int pageSize) throws SolrServiceException {
-		String topics = solrTopicService.searchTopics(query, fq, fl, facets, sort, page, pageSize);
-		if(StringUtils.isBlank(topics)) {
-			return "[]";
-		}
+	public SolrDocumentList searchTopics(String query, String fq, String fl, String facets, String sort, int page, int pageSize) throws SolrServiceException {
+		SolrDocumentList topics = solrTopicService.searchTopics(query, fq, fl, facets, sort, page, pageSize);
 		return topics;
 	}
 
+	public BaseTopicResultPage<?> buildResultsPage(Map<String,String[]> requestParams, SolrDocumentList solrResults) throws Exception {
```

Comment:
this method is too long, please refactor by extracting methods which build individual parts of the result page